src/App.jsx

```diff
@@ -1,33 +1,65 @@
 import './App.scss';
+import { useEffect, useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+const getFilteredMovies = (movies, query) => movies.filter((movie) => {
+  const formattedTitle = movie.title.toLowerCase();
+  const formattedDescription = movie.description.toLowerCase();
+  const formattedQuery = query.toLowerCase().trim();
+
+  return formattedTitle.includes(formattedQuery)
+    || formattedDescription.includes(formattedQuery);
+});
+
+export const App = () => {
+  const [searchValue, setSearchValue] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState([...moviesFromServer]);
+
+  useEffect(() => {
+    switch (searchValue) {
+      case '':
+        setVisibleMovies(moviesFromServer);
+        break;
+      default:
+        setVisibleMovies(getFilteredMovies(moviesFromServer, searchValue));
+    }
```

Comment:
Only for two conditions would the `?` structure be sufficient:
```suggestion
    const movies = searchValue ? moviesFromServer : getFilteredMovies(moviesFromServer, searchValue);
    setVisibleMovies(movies);
```